src/renderer/actions/notification.ts

```diff
@@ -0,0 +1,53 @@
+import {NotificationActionTypes} from "../constants/action-types";
+import {Action, Dispatch} from "redux";
+import {INotificationState} from "../reducers/notification";
+import {Horizontal, Vertical} from "../components/Notification/NotificationEnums";
+
+//Add Notification
+/** expireTime is in ms */
+export const storeNotificationAction = (notification: INotificationState, expireTime?: number
+) =>
+    (dispatch: Dispatch<IStoreNotificationAction>): void => {
+        let isStacked = false;
+        if(notification.horizontalPosition===Horizontal.RIGHT 
+            && notification.verticalPosition===Vertical.BOTTOM)
+            isStacked = true;
+        dispatch(setNotification( notification
+        ));
+        expireTime ? setTimeout(dispatch, expireTime, setRemoveNotification(0,isStacked)) : null;
```

Comment:
why are you always removing notification with index 0?

Reply:
Mistake, I was testing with buttons dispatching same action with same expireTime, so it worked perfectly only for that case, because latest notification was always with index 0. 

I'll fix it so it removes notification based on ID.